trunk/source/PEAtoBoogie/src/de/uni_freiburg/informatik/ultimate/pea2boogie/InvariantResultTransformer.java

```diff
@@ -0,0 +1,50 @@
+package de.uni_freiburg.informatik.ultimate.pea2boogie;
+
+import java.util.HashSet;
+import java.util.Set;
+
+import de.uni_freiburg.informatik.ultimate.boogie.ast.BinaryExpression;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.Expression;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.IdentifierExpression;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.UnaryExpression;
+
+public class InvariantResultTransformer {
+
+	private InvariantResultTransformer() {
+		// Utility
+		// This class could serve for more purposes in the future and also allow
+		// instances if needed but right now it's really just to have the extraction
+		// outside of other classes where it doesn't belong. In the future perhaps it could
+		// post-process for several reqcheck properties such as e.g., vacuity too.
+	}
+
+	public static Set<String> extractRedundancySet(final Expression invariant) {
+		final var redSet = new HashSet<String>();
+		extractRedundancySetHelper(invariant, redSet);
+		return redSet;
+	}
+
+	private static void extractRedundancySetHelper(final Expression expr, final HashSet<String> redSet) {
+		switch (expr) {
```

Comment:
I'm not sure this methods handles all kinds of expressions. It looks like we already have a class (`BoogieIdExtractor`) that you could use to extract the IDs, and then filter them afterwards.

Reply:
You are right, I use the already existing method for now. It's probably bad to create a new instance every time the method is called, rather than give the InvariantTransformer a proper constructor and a BoogieIdExtractor as a field, but for now this should work (unless you like the other option more).